services/port_manager/src/main/java/com/futurewei/alcor/portmanager/processor/NeighborProcessor.java

```diff
@@ -18,34 +18,31 @@
 import com.futurewei.alcor.portmanager.entity.PortNeighbors;
 import com.futurewei.alcor.portmanager.request.FetchPortNeighborRequest;
 import com.futurewei.alcor.portmanager.request.IRestRequest;
-import com.futurewei.alcor.web.entity.dataplane.InternalPortEntity;
 import com.futurewei.alcor.web.entity.dataplane.NeighborInfo;
 import com.futurewei.alcor.web.entity.port.PortEntity;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 import java.util.stream.Collectors;
 
 public class NeighborProcessor extends AbstractProcessor {
     private void fetchPortNeighborCallback(IRestRequest request) {
         List<PortNeighbors> portNeighborsList = ((FetchPortNeighborRequest) request).getPortNeighborsList();
-        List<InternalPortEntity> internalPortEntities =
-                request.getContext().getNetworkConfig().getPortEntities();
-
-        for (InternalPortEntity internalPortEntity : internalPortEntities) {
-            for (PortNeighbors portNeighbors: portNeighborsList) {
-                if (portNeighbors.getNeighbors() == null) {
-                    continue;
-                }
+        if (portNeighborsList == null || portNeighborsList.size() == 0) {
+            return;
+        }
 
-                if (internalPortEntity.getVpcId().equals(portNeighbors.getVpcId())) {
-                    List<NeighborInfo> neighborInfos = new ArrayList<>(portNeighbors.getNeighbors().values());
-                    internalPortEntity.setNeighborInfos(neighborInfos);
-                }
+        List<NeighborInfo> neighborInfos = new ArrayList<>();
+        for (PortNeighbors portNeighbors: portNeighborsList) {
+            if (portNeighbors.getNeighbors() == null ||
+                    portNeighbors.getNeighbors().size() == 0) {
+                continue;
             }
+
+            neighborInfos.addAll(portNeighbors.getNeighbors().values());
```

Comment:
What if neighborInfos contains some duplicated neighbors? Is it possible?

Reply:
The portNeighbors is read from the DB, and the they are stored with port_id as key, so the neighbor is not duplicated.